File: core-presentation/src/main/java/home/oleg/placesnearme/core_presentation/recyclerview/VenuesDiffCallback.java
package home.oleg.placesnearme.core_presentation.recyclerview;

import android.support.v7.util.DiffUtil;

import io.reactivex.annotations.NonNull;

public final class VenuesDiffCallback extends DiffUtil.ItemCallback<ItemViewType> {

    public static VenuesDiffCallback VENUES_DIFF_CALLBACK = new VenuesDiffCallback();

    private VenuesDiffCallback() {

    }

    @Override
    public boolean areItemsTheSame(@NonNull ItemViewType itemViewType, @NonNull ItemViewType t1) {
        return itemViewType.equals(t1);
    }

    @Override
    public boolean areContentsTheSame(@NonNull ItemViewType itemViewType, @NonNull ItemViewType t1) {
        return itemViewType.equals(t1);
    }

}

File: core-presentation/src/main/java/home/oleg/placesnearme/core_presentation/recyclerview/VenuesDiffCallback.kt
package home.oleg.placesnearme.core_presentation.recyclerview

import androidx.recyclerview.widget.DiffUtil
import io.reactivex.annotations.NonNull

class VenuesDiffCallback private constructor() : DiffUtil.ItemCallback<ItemViewType>() {

    override fun areItemsTheSame(itemViewType: ItemViewType, t1: ItemViewType): Boolean {
        return itemViewType == t1
    }

    override fun areContentsTheSame(itemViewType: ItemViewType, t1: ItemViewType): Boolean {
        return itemViewType == t1
    }

    companion object {

        var VENUES_DIFF_CALLBACK = VenuesDiffCallback()
    }

}
